### ISSUE TITLE
Canny causes crash

### ISSUE THREAD

**Issue by Akards** [JuliaImages/Images.jl#665]
Hi all,
Running the canny function causes the Julia console to crash. For example,
```
using Images
using TestImages 
img = testimage("mandrill")
img2 = canny(img)
```
would cause a crash.

For what it's worth, converting the image to grayscale or applying a filter doesn't help.

Thanks

**Comment by rdeits** [comment]
Can you please post the output of `versioninfo()` ? That will identify exactly which Julia version you're using. And can you also please post the output you see from Julia when it crashes?

**Comment by Akards** [comment]
versioninfo() returns this:
```
Julia Version 0.6.0
Commit 903644385b* (2017-06-19 13:05 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i3-5005U CPU @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, broadwell)
```
I'm not sure what the output from Julia is - I always have to close the console, and I usually have to reboot as well.

**Comment by Akards** [comment]
I should add that it hangs before it crashes, and never returns any text

**Comment by timholy** [comment]
Thanks for the bug report. [This line](https://github.com/JuliaImages/Images.jl/blob/ad827599cee1f279cf4067f38fc930a2ca5376fb/src/edge.jl#L405) is calling itself and thus causing a stackoverflow; it hangs because it's creating a new matrix each time so it's slow and looks like a hang. See #666.

**Comment by timholy** [comment]
I should mention that the way `canny` works now is that it requires you to pass the upper and lower thresholds: `canny(img, (hi, lo))`. You can use `Percentile` (e.g., `canny(img, (Percentile(90), Percentile(10)))`) for "rank-based" thresholds or just pass fixed numbers in.

It also seems like the docstring for `canny` hadn't been properly updated to the new API (sorry about that), so that is also fixed in #666.
